src/Veldrid/Vk/VkCommandList.cs

```diff
@@ -815,17 +815,23 @@ private protected override void UpdateBufferCore(DeviceBuffer buffer, uint buffe
 
             if (!sourceIsStaging && !destIsStaging)
             {
+                var srcAspect = (srcVkTexture.Usage & TextureUsage.DepthStencil) != 0
+                    ? VkImageAspectFlags.Depth | VkImageAspectFlags.Stencil
+                    : VkImageAspectFlags.Color;
                 VkImageSubresourceLayers srcSubresource = new VkImageSubresourceLayers
                 {
-                    aspectMask = VkImageAspectFlags.Color,
+                    aspectMask = srcAspect,
                     layerCount = layerCount,
                     mipLevel = srcMipLevel,
                     baseArrayLayer = srcBaseArrayLayer
                 };
 
+                var dstAspect = (dstVkTexture.Usage & TextureUsage.DepthStencil) != 0
```

Comment:
I only question whether `VkImageAspectFlags.Stencil` is permitted to be passed here if the texture has a depth-only format. There is at least one other place in the code that treats it separately: https://github.com/mellinoe/veldrid/blob/eef8375169d1960a322f47f95e9b6ee8126e7b43/src/Veldrid/Vk/VkFramebuffer.cs#L202. I suspect a validation error might be emitted if you use the Stencil flag but the texture is depth-only.

This would be confirmed one way or the other by adding a test that exercises this.